### PR TITLE
Installation instructions

### PR DESCRIPTION
**Pull request by AlexanderBand**

### CODE REVIEW

**Comment by AlexanderBand** [comment]
I think Ploutos currently still builds packages for Debian 9 and Ubuntu 18, but I don't explicitly mention those in the docs as we are likely to discontinue those.

File: README.md [hunk 1]
```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
...
+compatible systems such as Rocky Linux. 
+
+You can also build `dnsi` from the source code using Cargo, Rust's build
+system and package manager. Cargo lets you to run `dnsi` on almost any
+operating system and CPU architecture. Refer to the [building](#building)
+section to get started.
+
+### Debian
+
+To install a `dnsi` package, you need the 64-bit version of one of these
```
Comment (tertsdiepraam) [inline README.md:29]:
```suggestion
To install the `dnsi` package, you need the 64-bit version of one of these
```

File: README.md [hunk 2]
```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
...
+Debian versions:
+
+-  Debian Bookworm 12
+-  Debian Bullseye 11
+-  Debian Buster 10
+
+Packages for the `amd64`/`x86_64` architecture are available for
```
Comment (tertsdiepraam) [inline README.md:36]:
```suggestion
Packages for the `amd64` and  `x86_64` architectures are available for
```

File: README.md [hunk 3]
```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
...
+
+You can now install `dnsi` with:
+
+``` bash
+sudo apt install dnsi
+```
+
+### RHEL and compatible systems
+
+To install a `dnsi` package, you need Red Hat Enterprise Linux (RHEL) 7,
```
Comment (tertsdiepraam) [inline README.md:136]:
```suggestion
To install the `dnsi` package, you need Red Hat Enterprise Linux (RHEL) 7,
```

File: README.md [hunk 4]
```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
...
+Alternatively, visit the [Rust website](https://www.rust-lang.org/tools/install) for other installation methods. Also refer to this page for notes
+on updating Rust, and configuring the `PATH` environment variable.
+
+### Installing and updating `dnsi`
+
+After successfully installing Rust, installing `dnsi` is as simple as
+entering:
+
+``` bash
+cargo install dnsi
```
Comment (tertsdiepraam) [inline README.md:193]:
```suggestion
cargo install --locked dnsi
```
I think this helps a bit if some dependency breaks semver. It's a good extra guard to have, I think.

File: README.md [hunk 5]
```diff
@@ -12,3 +12,205 @@ The tool contains a number of commands. Currently, these are:
...
+
+Use the `--force` option to overwrite an existing version with the latest
+`dnsi` release:
+
+``` bash
+cargo install --locked --force dnsi
+```
+
+If you want to install a specific version of `dnsi` using Cargo, explicitly
+use the ``--version`` option. If needed, use the ``--force`` option to
```
Comment (tertsdiepraam) [inline README.md:211]:
```suggestion
use the `--version` option. If needed, use the `--force` option to
```

File: README.md (hunk 4 above)
Comment (ximon18) [reply README.md:193]: The downside of using locked ia that installers won't benefit from supposedly compatible fixes to dependencies used by dnsi.

Comment (ximon18) [reply README.md:193]: One other reason to use locked is that is how Ploutos builds binaries for inclusion in packages and thus anyone installing using locked gets the same versions as if they install from a package built by us.

**Comment by partim** [comment]
Given that these instruction include as yet not existing binary packages, we shall merge it just before the next release.

**Review by partim — APPROVED** [review]

**Comment by ximon18** [comment]
> I think Ploutos currently still builds packages for Debian 9 and Ubuntu 18, but I don't explicitly mention those in the docs as we are likely to discontinue those.

That's configurable per project. See: https://github.com/NLnetLabs/dnsi/blob/main/pkg/rules/packages-to-build.yml

**Pull request merged**